File: backend/MemeCreator.Api/MemeCreator.Domain/Entities/MemeConfig.cs
namespace MemeCreator.Domain.Entities;

public class MemeConfig
{
    public int Id { get; set; }

    public string TopText { get; set; } = string.Empty;
    public string BottomText { get; set; } = string.Empty;

    public string FontFamily { get; set; } = "Impact";
    public int FontSize { get; set; }

    public string TextColor { get; set; } = "#FFFFFF";
    public string StrokeColor { get; set; } = "#000000";
    public int StrokeWidth { get; set; }

    public string TextAlign { get; set; } = "center";
    public int Padding { get; set; }
    public bool AllCaps { get; set; }

    public byte[]? WatermarkImage { get; set; }
    public string? WatermarkPosition { get; set; }

    public float ScaleDown { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/models/MemeConfig.ts
export type TextAlign = "left" | "center" | "right";
export type WatermarkPosition =
  | "top-left"
  | "top-right"
  | "bottom-left"
  | "bottom-right";

export interface MemeConfig {
  topText: string;
  bottomText: string;
  fontFamily: string;
  fontSize: number;
  textColor: string;
  strokeColor: string;
  strokeWidth: number;
  textAlign: TextAlign;
  padding: number;
  allCaps: boolean;
  scaleDown: number;

  watermarkImageBase64?: string | null;
  watermarkPosition?: WatermarkPosition;
}
